pyteal/ast/abi/tuple.py

```diff
@@ -194,15 +194,10 @@ def _index_tuple(
     start_index = Int(offset)
     length = Int(valueType.byte_length_static())
 
-    if index + 1 == len(value_types):
-        if offset == 0:
+    if offset == 0:
+        if index + 1 == len(value_types):
             # This is the first and only value in the tuple, so decode all of encoded
             return output.decode(encoded)
-        # This is the last value in the tuple, so decode the substring from start_index to the end of
-        # encoded
-        return output.decode(encoded, start_index=start_index)
```

Comment:
I think this case is still correct for tuples of all static types. Do you think we should keep it for that case?